catkin_tools/jobs/utils.py

```diff
@@ -50,42 +50,67 @@ def get_env_loaders(package, context):
     return sources
 
 
+def merge_envs(job_env, overlay_envs):
+    '''
+    In the merged/linked case of single env, this function amounts to a straight
+    assignment, but a more complex merge is required with isolated result spaces,
+    since a package's build environment may require extending that of multiple
+    other result spaces.
+    '''
+    merge_path_values = {}
+
+    for overlay_env in overlay_envs:
+        for key, values_str in overlay_env.items():
+            if key.endswith('PATH'):
+                if key not in merge_path_values:
+                    # Seed the list with any values already in the environment. We reverse the list
+                    # here so that we can cheaply append to it, representing a prepend in the final
+                    # PATH var, and because we need to maintain the order of underlay paths.
+                    if key in job_env:
+                        values = job_env[key].split(':')
```

Comment:
I have no delusions that this will work on Windows, but there is a placeholder for this in Python that is portable:

https://docs.python.org/2.7/library/os.html#os.pathsep